display_list/display_list_unittests.cc

```diff
@@ -2499,6 +2504,296 @@ TEST(DisplayList, RTreeOfSaveLayerFilterScene) {
+
+    DisplayListBuilder builder2;
+    builder2.save();
+    builder2.transform(SkM44::Translate(10, 100));
+    builder2.drawRect({0, 0, 100, 100});
+    builder2.restore();
+    builder2.save();
+    builder2.transform(SkM44::Translate(10, 100));
+    builder2.drawRect({0, 0, 100, 100});
+    builder2.restore();
+    auto display_list2 = builder2.Build();
+
+    ASSERT_TRUE(DisplayListsEQ_Verbose(display_list1, display_list2));
+  }
+  // Skew
+  {
+    DisplayListBuilder builder1;
+    builder1.save();
+    builder1.save();
+    builder1.skew(10, 10);
+    builder1.drawRect({0, 0, 100, 100});
+    builder1.restore();
+    auto display_list1 = builder1.Build();
+
+    DisplayListBuilder builder2;
+    builder2.save();
+    builder2.skew(10, 10);
+    builder2.drawRect({0, 0, 100, 100});
+    builder2.restore();
+    auto display_list2 = builder2.Build();
+
+    ASSERT_TRUE(DisplayListsEQ_Verbose(display_list1, display_list2));
+  }
+
+  // Scale
+  {
+    DisplayListBuilder builder1;
+    builder1.save();
+    builder1.save();
+    builder1.scale(0.5, 0.5);
+    builder1.drawRect({0, 0, 100, 100});
+    builder1.restore();
+    auto display_list1 = builder1.Build();
+
+    DisplayListBuilder builder2;
+    builder2.save();
+    builder2.scale(0.5, 0.5);
+    builder2.drawRect({0, 0, 100, 100});
+    builder2.restore();
+    auto display_list2 = builder2.Build();
+
+    ASSERT_TRUE(DisplayListsEQ_Verbose(display_list1, display_list2));
+  }
+
+  {
+    DisplayListBuilder builder1;
+    builder1.save();
+    builder1.save();
+    builder1.transform(SkM44());
+    builder1.drawRect({0, 0, 100, 100});
+    builder1.restore();
+    builder1.transform(SkM44());
+    builder1.drawRect({0, 0, 100, 100});
+    builder1.restore();
+    auto display_list1 = builder1.Build();
+
+    DisplayListBuilder builder2;
+    builder2.drawRect({0, 0, 100, 100});
+    builder2.drawRect({0, 0, 100, 100});
+    auto display_list2 = builder2.Build();
+
+    ASSERT_TRUE(DisplayListsEQ_Verbose(display_list1, display_list2));
+  }
+  // Clip
+  {
+    DisplayListBuilder builder1;
+    builder1.save();
+    builder1.save();
+    builder1.clipRect(SkRect::MakeLTRB(0, 0, 100, 100), SkClipOp::kIntersect,
+                      true);
+    builder1.drawRect({0, 0, 100, 100});
+    builder1.restore();
+    builder1.transform(SkM44());
+    builder1.drawRect({0, 0, 100, 100});
+    builder1.restore();
+    auto display_list1 = builder1.Build();
+
+    DisplayListBuilder builder2;
+    builder2.save();
+    builder2.clipRect(SkRect::MakeLTRB(0, 0, 100, 100), SkClipOp::kIntersect,
+                      true);
+    builder2.drawRect({0, 0, 100, 100});
+    builder2.restore();
+    builder2.transform(SkM44());
+    builder2.drawRect({0, 0, 100, 100});
+    auto display_list2 = builder2.Build();
+
+    ASSERT_TRUE(DisplayListsEQ_Verbose(display_list1, display_list2));
+  }
+
+  std::unordered_set<std::string> function_names = {"Translate",
```

Comment:
This loop should be moved into its own TEST - named something like "TransformClipTestInvokersTriggerDeferredSave"?